Makefile

```diff
@@ -83,18 +93,23 @@ generate: generate-go generate-deploy generate-test generate-external
 check: $(GO)
 	./hack/check.sh
 
-manager: $(GO)
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GO) build -o $(BIN_DIR)/manager github.com/k8snetworkplumbingwg/kubemacpool/cmd/manager
-
 # Build the docker image
-container: manager
-	$(OCI_BIN) build build/ -t ${REGISTRY}/${IMG}:${IMAGE_TAG}
+container:
+ifeq ($(OCI_BIN),podman)
+	$(MAKE) build-multiarch-kubemacpool-podman
+else ifeq ($(OCI_BIN),docker)
+	$(MAKE) build-multiarch-kubemacpool-docker
+else
+	$(error Unsupported OCI_BIN value: $(OCI_BIN))
+endif
 
 # Push the docker image
 docker-push:
-	$(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_TAG}
-	$(OCI_BIN) tag ${REGISTRY}/${IMG}:${IMAGE_TAG} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}
-	$(OCI_BIN) push ${TLS_SETTING} ${REGISTRY}/${IMG}:${IMAGE_GIT_TAG}
+ifeq ($(OCI_BIN),podman)
```

Comment:
why is there no docker counterpart? 
In any case, if we don't want to support pushing with docker, we should be explicit, failing the operation.

Reply:
Let me rephrase: in the existing code (https://github.com/k8snetworkplumbingwg/kubemacpool/blob/main/Makefile#L94-L98) you can either do docker/podman push/tag using the OCI_BIN env var.
Now however, you can only do it with podman:
```
ifeq ($(OCI_BIN),podman)
    push,tag,etc...
endif
...
```

my original question is - why are you limiting it like this?

Reply:
For building multi-platform images using docker buildx build, the build and push operations are combined into a single command, Therefore, the separate push and tag logic, which currently applies only to podman, isn't needed for docker when using buildx.

Reply:
theoretically it is possible to split even for docker single arch / multi arch ?
(i see here `--push` flag is used so i wonder)
https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441/files#diff-8a6127c87c24bc27e5595386a152b920da0266e3b6d185a3e632cab440abf09bR10-R18

not saying we need to, first trying to understand

Reply:
Currently it's not possible for multi arch images:
Below is the reason:

With docker buildx, the build result is stored only in the build cache by default. There are two options to handle the result, which I've already tested: --push and --load.

The --push flag pushes the built image to a container registry.
The --load flag loads the resulting image into the local Docker daemon. However, the --load option currently only supports single-platform images or simpler images. It does not work with multi-platform builds (i.e., manifest lists).
When using the --load flag with a multi-platform build, you'll encounter the following error:

docker exporter does not currently support exporting manifest lists

Reply:
thanks for the explanation
theoretically for single arch, it is possible then right?
it is nice to have, as it will allow to just build instead build + push, but i am not sure it worths the effort, and all the other repos have it combined already
moreover we encourage podman over docker

it is Ram decision here

Reply:
Correct, but with single arch we can directly use docker build , for multiplatform we need to use docker buildx build and we have to do like this

Reply:
I see now. I don't like the asymmetry of it, but having build+push to the local repo isn't such a bad thing. 
Thanks for the explanation